src/uct/ib/dc/base/dc_iface.c

```diff
@@ -132,6 +132,23 @@ static ucs_status_t uct_dc_iface_dci_connect(uct_dc_iface_t *iface, uct_rc_txqp_
     return UCS_OK;
 }
 
+/* restore dc qp from error to rts state */
+ucs_status_t uct_dc_iface_dci_reconnect(uct_dc_iface_t *iface,
+                                        uct_rc_txqp_t *dci)
+{
+    struct ibv_exp_qp_attr attr;
+
+    memset(&attr, 0, sizeof(attr));
+    attr.qp_state = IBV_QPS_RESET;
+
+    if (ibv_exp_modify_qp(dci->qp, &attr, IBV_EXP_QP_STATE)) {
```

Comment:
pls take a look on rc_mlx5_ep.c:466:
1. need to reuse code from uct_rc_ep_reset_qp (provavly create common function to accept uct_rc_txqp_t*)
2. for mlx5, need to call uct_ib_mlx5_update_cq_ci() before, and update cq_ci after, the call to reset. this is because the driver can change internal cq->cons_index when modifying qp to reset and need to resync it. pls extract common code for that with rc_mlx5, if possible.